### PR TITLE
[TS] LPS-166867 | Cannot edit document in Office 365 when file in OneDrive has been sent to trash

### PR DESCRIPTION
**Pull request by ricardocousolr**
Hi @liferay-lima!

This PR is to fix a possible inconsistency situation in the integration with Office 365, as described in [LPS-166867](https://issues.liferay.com/browse/LPS-166867).

The idea of the solution is to treat the exception `GraphServicePortalException.ItemNotFound` in a special way (as it kind of is in `DLOpenerOneDriveDLAppServiceWrapper#cancelCheckOut`) by deleting the reference in `DeleteDLOpenerFileEntryReference` so that it goes in sync with the checking out of the File Entry (that updates/removes the PWC in `DLFileVersion`).

Please let me know if you have any questions. 

Thanks!

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Review by AliciaGarciaGarcia — APPROVED** [review]

File: modules/dxp/apps/document-library-opener/document-library-opener-onedrive-web/src/main/java/com/liferay/document/library/opener/onedrive/web/internal/DLOpenerOneDriveManager.java [hunk 1]
```diff
@@ -138,6 +139,19 @@ public void deleteFile(long userId, FileEntry fileEntry)
...
+					GraphServicePortalException.ItemNotFound) {
+
+				_dlOpenerFileEntryReferenceLocalService.
+					deleteDLOpenerFileEntryReference(
+						DLOpenerOneDriveConstants.ONE_DRIVE_REFERENCE_TYPE,
+						fileEntry);
+			}
+
 			throw GraphServiceExceptionPortalExceptionMapper.map(
 				graphServiceException);
```
Comment (adolfopa) [inline modules/dxp/apps/document-library-opener/document-library-opener-onedrive-web/src/main/java/com/liferay/document/library/opener/onedrive/web/internal/DLOpenerOneDriveManager.java:156]: Why are you calling the `map` method twice? Shouldn't you simply `throw graphServicePortalException`?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/document-library-opener/document-library-opener-onedrive-web/src/main/java/com/liferay/document/library/opener/onedrive/web/internal/DLOpenerOneDriveManager.java (hunk 1 above)
Comment (ricardocousolr) [reply modules/dxp/apps/document-library-opener/document-library-opener-onedrive-web/src/main/java/com/liferay/document/library/opener/onedrive/web/internal/DLOpenerOneDriveManager.java:156]: You're absolutely right. I'll fix that.

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**